### PR TITLE
Add managed identity support

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**
Adding MI support for access to service bus topics. Needs testing.
Please suggest any improvements.

### CODE REVIEW

File: managed-identity.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+locals {
+  managed_identity_list = toset(compact(concat(var.managed_identity_object_ids, [var.managed_identity_object_id])))
+}
+
+resource "azurerm_user_assigned_identity" "managed_identity" {
+  resource_group_name = "managed-identities-${var.env}-rg"
+  location            = var.location
+  name                = "${var.product}-${var.env}-mi"
+
+  count = var.create_managed_identity ? 1 : 0
```
Comment (timja) [inline managed-identity.tf:10]: https://github.com/hmcts/terraform-module-servicebus-subscription/pull/23#discussion_r1340104282

Comment (nikola-naydenov-hmcts) [reply managed-identity.tf]: Removed.

File: managed-identity.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+locals {
+  managed_identity_list = toset(compact(concat(var.managed_identity_object_ids, [var.managed_identity_object_id])))
+}
+
+resource "azurerm_role_assignment" "mi_role_assignment_sender" {
+  principal_id         = each.value
+  role_definition_name = "Azure Service Bus Data Sender"
```
Comment (timja) [inline managed-identity.tf:7]: There's no need for the receiver role? i.e. should this be part of the DSL? maybe a list objects with id and roles, or name and roles?

Comment (nikola-naydenov-hmcts) [reply managed-identity.tf:7]: I am going off this explanation from their docs:

https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-managed-service-identity

[Azure Service Bus Data Owner](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#azure-service-bus-data-owner): Use this role to allow full access to Service Bus namespace and its entities (queues, topics, subscriptions, and filters)
[Azure Service Bus Data Sender](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#azure-service-bus-data-sender): Use this role to allow sending messages to Service Bus queues and topics.
[Azure Service Bus Data Receiver](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#azure-service-bus-data-receiver): Use this role to allow receiving messages from Service Bus queues and subscriptions.

**Review by timja — APPROVED** [review]

**Pull request merged**